cl/scrapers/management/commands/clone_from_cl.py

```diff
@@ -0,0 +1,298 @@
+            court_datum = s.get(docket_datum["court"]).json()
+            citation_data = cluster_datum["citations"]
+            opinion_data = cluster_datum["sub_opinions"]
+            # delete resource_uri value generated by DRF
+            del court_datum["resource_uri"]
+            del docket_datum["resource_uri"]
+            del cluster_datum["resource_uri"]
+            # delete fields with fk or m2m relations or unneeded fields
+            del docket_datum["clusters"]
+            del docket_datum["original_court_info"]
+            del docket_datum["absolute_url"]
+            del docket_datum["audio_files"]
+            del docket_datum["tags"]
+            del docket_datum["panel"]
+            del cluster_datum["docket"]
+            del cluster_datum["citations"]
+            del cluster_datum["sub_opinions"]
+            del cluster_datum["absolute_url"]
+            del cluster_datum["panel"]
+            del cluster_datum["non_participating_judges"]
+            with transaction.atomic():
+                # Get or create required court
+                try:
+                    ct = Court.objects.get_or_create(**court_datum)
+                except:
+                    ct = Court.objects.filter(pk=court_datum["id"])
+                docket_datum["court"] = ct[0]
+
+                # Create docket
+                Docket.objects.create(**docket_datum)
+
+                for cite_data in citation_data:
+                    cite_data["cluster_id"] = cluster_datum["id"]
+                cluster_datum["docket_id"] = docket_datum["id"]
+
+                # Create opinion cluster
+                OpinionCluster.objects.create(**cluster_datum)
+
+                for cite_data in citation_data:
+                    # Create citations
+                    cite_data["cluster_id"] = cluster_datum["id"]
+                    Citation.objects.create(**cite_data)
+
+                for op in opinion_data:
+                    op_data = s.get(op).json()
+                    # delete fields with fk or m2m relations or unneeded fields
+                    del op_data["opinions_cited"]
+                    del op_data["cluster"]
+                    del op_data["absolute_url"]
+                    del op_data["resource_uri"]
+                    del op_data["author"]
+                    del op_data["joined_by"]
+                    op_data["cluster_id"] = cluster_datum["id"]
+
+                    # Create opinion
+                    op = Opinion.objects.create(**op_data)
+
+                    # Add opinion to search engine
+                    add_items_to_solr.delay([op.id], "search.Opinion")
+
+                print(
+                    f"http://localhost:8000/opinion/{cluster_datum['id']}/{docket_datum['slug']}/"
+                )
+
+    def clone_docket(self) -> None:
+        """Download docket data from courtlistener.com and add it to local
+        environment
+        """
```

Comment:
Could you refactor this and the other methods so they are called from the clone_opinion method, instead of repeating code?